src/compiler/checker.ts

```diff
@@ -2452,6 +2452,69 @@ module ts {
             return result;
         }
 
+        function getBaseTypes(type: InterfaceType): ObjectType[] {
+            if (!(<InterfaceTypeWithBaseTypes>type).baseTypes) {

```

Comment:
you do the cast 4 times.  Can you just cast once to a local, and then use that from then on?